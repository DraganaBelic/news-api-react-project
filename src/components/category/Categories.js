import React from "react";
import { useSelector } from "react-redux";
import { CategoryItemWrapper } from "./CategoryStyle";
import CategoryExpandableItem from "./CategoryExpandableItem";
import { categories } from "../../constants/Constant";
import { useTranslation } from "react-i18next";

const Categories = () => {
  const { t } = useTranslation();
  const currentCountry = useSelector((state) => state.country.currentCountry);
  return (
    <div>
      <h2>
        {t("top5NewsByCategoryFrom")} {t(currentCountry)}
      </h2>
      <CategoryItemWrapper>
        {Object.values(categories).map((category, i) => {
          return <CategoryExpandableItem key={i} category={category} />;
        })}
      </CategoryItemWrapper>
    </div>
  );
};

export default Categories;
