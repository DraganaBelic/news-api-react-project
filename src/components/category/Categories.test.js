import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import thunk from "redux-thunk";
import { MemoryRouter } from "react-router-dom";
import Categories from "./Categories";

const middlewares = [thunk];
const initialState = {
  country: {
    currentCountry: "gb",
  },
};
export const mockedStore = (state = initialState) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};
jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (s) => s }),
}));
describe("Test Categories component", () => {
  it("should render with given state from Redux store", () => {
    const store = mockedStore();
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Categories />
        </MemoryRouter>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
