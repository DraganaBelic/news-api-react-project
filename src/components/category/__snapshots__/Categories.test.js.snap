// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Categories component should render with given state from Redux store 1`] = `
<div>
  <h2>
    top5NewsByCategoryFrom
     
    gb
  </h2>
  <div
    class="CategoryStyle__CategoryItemWrapper-yyfh8s-0 Ltyni"
  >
    <div>
      <div
        class="CategoryStyle__CategoryTitleStyle-yyfh8s-2 cbhErF"
      >
        <a
          href="/gb/entertainment"
        >
          entertainment
           
        </a>
        <button>
          <i
            class="fas fa-chevron-down"
          />
        </button>
      </div>
      <div
        class="CategoryStyle__NewsByCategoryWrapperStyle-yyfh8s-1 kJCvmv"
      />
    </div>
    <div>
      <div
        class="CategoryStyle__CategoryTitleStyle-yyfh8s-2 cbhErF"
      >
        <a
          href="/gb/general"
        >
          general
           
        </a>
        <button>
          <i
            class="fas fa-chevron-down"
          />
        </button>
      </div>
      <div
        class="CategoryStyle__NewsByCategoryWrapperStyle-yyfh8s-1 kJCvmv"
      />
    </div>
    <div>
      <div
        class="CategoryStyle__CategoryTitleStyle-yyfh8s-2 cbhErF"
      >
        <a
          href="/gb/health"
        >
          health
           
        </a>
        <button>
          <i
            class="fas fa-chevron-down"
          />
        </button>
      </div>
      <div
        class="CategoryStyle__NewsByCategoryWrapperStyle-yyfh8s-1 kJCvmv"
      />
    </div>
    <div>
      <div
        class="CategoryStyle__CategoryTitleStyle-yyfh8s-2 cbhErF"
      >
        <a
          href="/gb/science"
        >
          science
           
        </a>
        <button>
          <i
            class="fas fa-chevron-down"
          />
        </button>
      </div>
      <div
        class="CategoryStyle__NewsByCategoryWrapperStyle-yyfh8s-1 kJCvmv"
      />
    </div>
    <div>
      <div
        class="CategoryStyle__CategoryTitleStyle-yyfh8s-2 cbhErF"
      >
        <a
          href="/gb/technology"
        >
          technology
           
        </a>
        <button>
          <i
            class="fas fa-chevron-down"
          />
        </button>
      </div>
      <div
        class="CategoryStyle__NewsByCategoryWrapperStyle-yyfh8s-1 kJCvmv"
      />
    </div>
  </div>
</div>
`;
